### PR TITLE
fix: use distinct ids for cohort exports

### LINKED ISSUE #23462: Bug report: Cohorts - Exporting omits the ID seen in the `Person` column
## Bug description

When on a cohort page:

> When I click 'export' and 'current columns' on this page, the person ID is suddenly the same as the ID. Therefore I am unable to use this to match the data back to our internal system which has Person ID. I attach the export for your reference.

CleanShot 2024-07-03 at 16 42 55@2x

Example cohort: https://us.posthog.com/project/60913/cohorts/76772

## How to reproduce

1. Create or view a cohort
2. Export current columns to a csv file
3. Open the file
4. Compare the IDs in the first two columns with the IDs displayed in the first two columns of the UI
5. Notice that the ID displayed the leftmost column (`Persons`) in the UI, is replaced by the ID displayed in the `ID` column of the UI, as in the screenshot above

The ID displayed the leftmost column (`Persons`) in the UI is also not found in the exported file when exporting "All columns"

## Additional context

From: https://posthoghelp.zendesk.com/agent/tickets/15369

### Debug info

```shell
Admin: http://go/adminOrgUS/018c7a63-e197-0000-c80b-c72916710820 (Project: 60913)
Sentry: http://go/sentryUS/60913
```

### PR DESCRIPTION
**Pull request by derek-li**
## Problem

Fixes https://github.com/PostHog/posthog/issues/23462

When on a cohort page, clicking export results in duplicate columns

## Changes

ids are now consistent:
SCR-20240723-mcdx
SCR-20240723-mcga

## Does this work well for both Cloud and self-hosted?

n/a

## How did you test this code?

Checked new export files against in-app table contents

### CODE REVIEW

**Comment by derek-li** [comment]
Hey @slshults, this is my first time contributing. How do I get eyes on this for review?

Tagging you because you created the issue. Thanks!

**Comment by slshults** [comment]
> Hey @slshults, this is my first time contributing. How do I get eyes on this for review?
> 
> Tagging you because you created the issue. Thanks!

Hey @derek-li , Very cool, thank you!  

I've requested a review from one of our product analytics engineers, @anirudhpillai .

**Comment by slshults** [comment]
Btw, @derek-li , could you please send me an email, steven @ posthog.com ? I'd like to send you some credit for our merch store! 🙂

File: frontend/src/queries/nodes/DataTable/DataTableExport.tsx [hunk 1]
```diff
@@ -54,7 +54,9 @@ export async function startDownload(
             )
         }
         if (exportContext['columns'].includes('person')) {
-            exportContext['columns'] = exportContext['columns'].map((c: string) => (c === 'person' ? 'person.id' : c))
+            exportContext['columns'] = exportContext['columns'].map((c: string) =>
```
Comment (anirudhpillai) [inline frontend/src/queries/nodes/DataTable/DataTableExport.tsx:57]: Thanks a lot for the changes Derek!
Any reason you name the column 'person.distinct_ids.0' over `person.distinct_id`?

Comment (derek-li) [reply frontend/src/queries/nodes/DataTable/DataTableExport.tsx:57]: @anirudhpillai `distinct_id` does not exist as a property of `person`. The back end is doing some special logic (not 100% clear on what's happening) to query all `distinct_ids` for a user, but csvs are not great at handling lists which is why I pick the first one 

Here's what happens if we try `person.distinct_id`:
image

**Review by anirudhpillai — APPROVED** [review]
Looks good to me. The first column won't always show the first distinct id on the UI. It will sometimes show the email...., the logic for that is [here](https://github.com/PostHog/posthog/blob/master/frontend/src/scenes/persons/PersonDisplay.tsx#L32). Example is the test data locally
But this is still useful functionality to have the distinct id there so let's get this in.
image

**Pull request merged**